common/djangoapps/edraak_i18n/helpers.py

```diff
@@ -0,0 +1,54 @@
+"""
+Helper functions to Edraak i18n module.
+"""
+
+
+def add_locale_middleware(middleware_classes):
+    """
+    Adds the Edraak's DefaultLocaleMiddleware to the MIDDLEWARE_CLASSES tuple correctly.
+
+    This function is meant to be used within the settings files like lms/envs/aws.py and others.
+
+    Args:
+        middleware_classes: The MIDDLEWARE_CLASSES tuple from the settings.
+
+    Returns:
+        The new MIDDLEWARE_CLASSES with the edraak_i18n middleware.
+    """
+
+    edraak_middleware = 'edraak_i18n.middleware.DefaultLocaleMiddleware'
+
+    other_locale_middlewares = (
+        'lang_pref.middleware.LanguagePreferenceMiddleware',
+        'dark_lang.middleware.DarkLangMiddleware',
+        'django.middleware.locale.LocaleMiddleware',
+    )
+
+    indexes = [
+        middleware_classes.index(class_name) for class_name in other_locale_middlewares
+        if class_name in middleware_classes
+    ]
+
+    first_index = min(indexes)
+
+    # Insert the DefaultLocaleMiddleware any other locale-related middleware in order for it to work
+    return middleware_classes[:first_index] + (edraak_middleware,) + middleware_classes[first_index:]
```

Comment:
But here you are not checking for ```SessionMiddleware``` or any other middleware that Locale middleware depends on, thus defensive programming does not exactly apply here, maybe if there was a function that checks middlewares and rearranges them correctly, what do you think?